### ISSUE TITLE
AssertionError: HEUDICONV_VERSION_JSON_KEY not in json_ 

### ISSUE THREAD

**Issue by bpinsard** [nipy/heudiconv#602]
### Summary

I got the following error on some pilot data
```
Traceback (most recent call last):                                                                                                                        
  File "/data/neuromod/virtualenvs/neuromod_data/bin/heudiconv", line 8, in <module>                                                                      
    sys.exit(main())                                                                                                                                      
  File "/data/neuromod/virtualenvs/neuromod_data/lib/python3.7/site-packages/heudiconv/cli/run.py", line 24, in main                                      
    workflow(**kwargs)                                                                                                                                    
  File "/data/neuromod/virtualenvs/neuromod_data/lib/python3.7/site-packages/heudiconv/main.py", line 393, in workflow                                    
    grouping=grouping,)                                                                                                                                   
  File "/data/neuromod/virtualenvs/neuromod_data/lib/python3.7/site-packages/heudiconv/convert.py", line 221, in prep_conversion                          
    dcmconfig=dcmconfig,)                                                                                                                                 
  File "/data/neuromod/virtualenvs/neuromod_data/lib/python3.7/site-packages/heudiconv/convert.py", line 548, in convert                                  
    tuneup_bids_json_files(bids_outfiles)                                                                                                                 
  File "/data/neuromod/virtualenvs/neuromod_data/lib/python3.7/site-packages/heudiconv/bids.py", line 278, in tuneup_bids_json_files                      
    assert HEUDICONV_VERSION_JSON_KEY not in json_                                                                                                        
AssertionError 
```

Which is weird, as it seems that heudiconv insert that value in the json then checks that it's not there.
Any clue about what is going on?

### Platform details:

Choose one:
- [x] Local environment

- [ ] Container

- Heudiconv version: 

this is the branch enh-custom_seqinfo though it seems unrelated to these changes:
0.11.3.post9+gc939bf8

**Comment by yarikoptic** [comment]
the assumption of that `assert` was that we would never "tune up" the same json file over again. So we need to figure out how you got there.  

- Are you running with `--overwrite`?  (better share full invocation)
- run with `--dbg` and you would see on which file it does it. if you have git history for it -- can check when was populated and then we would see if it is just a wrong assumption of that assert (I think we should allow in `--overwrite` mode)

**Comment by bpinsard** [comment]
I must admit this is a very particular case, CMRR multi-band multi-echo mag+phase data 🤯  so each series will output (SBRef+BOLD)*(mag+phase)*(3 echoes) =12 files . On top of that, we acquired reversed-phase short series for distortion correction.
Ran it with a non-existing dataset, so it is not overwriting existing data ( `--overwrite` gives the same error) 
`heudiconv --dbg -b -l pilot -f /path/to/heuristics.py --files /path/to/dicoms`
The error is very likely in the heuristics, will trace all the seqinfos generated to find the source of the error. 
Thanks for the pointers, will update when I found my error.
